src/Term/TermFallback.php

```diff
@@ -46,6 +46,18 @@ public function __construct( $requestedLanguageCode, $text, $actualLanguageCode,
 			throw new InvalidArgumentException( '$sourceLanguageCode must be a non-empty string or null' );
 		}
 
+		if ( $actualLanguageCode === $sourceLanguageCode ) {
```

Comment:
I suggest to implement well tested `isFallback` and `isTransliteration` (and maybe `isVariant`) methods instead of throwing exceptions. In the given example all these methods will return `false`. So yes, this means you can have a TermFallback object that is *not* a fallback. I find this much more convenient: Code that constructs Term objects does not need complicated `if` constructs, it can be straight and create TermFallback objects for all terms, no matter how the given fallback actually resolves for each of them.